### PR TITLE
Make boost::filesystem optional if we have c++17

### PR DESCRIPTION
**Pull request by constantinpape**
This turned out to be a bit more complicated due to std::experimental::filesystem in gcc7.

### CODE REVIEW

**Comment by constantinpape** [comment]
@stuarteberg this should do the trick.
Fixes #124.

**Comment by constantinpape** [comment]
Fixes #72.
However, the tests in appveyor take very long and go over the time limit if all are run.
Only running a subset for now, see also #125.

**Comment by stuarteberg** [comment]
Thanks!  I'll try out the new version once conda-forge/z5py-feedstock#18 is ready.

File: include/z5/attributes.hxx [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
-namespace fs = boost::filesystem;
+#ifdef WITH_BOOST_FS
+    namespace fs = boost::filesystem;
+#else
+    #if __GCC__ > 7
+        namespace fs = std::filesystem;
+    #else
+        namespace fs = std::experimental::filesystem;
+    #endif
+#endif
```
Comment (jakirkham) [inline include/z5/attributes.hxx]: Maybe it makes sense to pull this out into some utility header and include the header everywhere. WDYT?

Comment (constantinpape) [reply include/z5/attributes.hxx]: Definitely. Thought about that too while refactoring, but wanted to test it first and then forgot about it.

**Pull request merged**